.drone.yml

```
@@ -1,297 +1,391 @@
+    environment:
+      DB_VENDOR: h2
+  - name: postgres
+    image: quay.io/ukhomeofficedigital/postgres-alpine
+    environment:
+      POSTGRES_USER: root
+      POSTGRES_PASSWORD: dev
+
+environment:
+  KEYCLOAK_SERVER_ROOT: http://keycloak:8080
+  KEYCLOAK_USER: admin
+  KEYCLOAK_PASSWORD: password1
+
+steps:
+
+  - name: wait for keycloak
+    image: quay.io/ukhomeofficedigital/hocs-docker-tools
     commands:
     - |
-      until $(curl --output /dev/null --silent --head --fail http://keycloak:8080/auth/realms/master/.well-known/openid-configuration); do
+      until
+        curl
+        --output /dev/null
+        --silent --head --fail
+        ${KEYCLOAK_SERVER_ROOT}/auth/realms/master/.well-known/openid-configuration
+      ; do
         echo "waiting for keycloak"
-        sleep 5
+        sleep 1
       done
-    - |
-      export TKN=$(curl -X POST 'http://keycloak:8080/auth/realms/master/protocol/openid-connect/token' -H "Content-Type: application/x-www-form-urlencoded" -d "username=admin" -d 'password=password1' -d 'grant_type=password' -d 'client_id=admin-cli' | jq -r '.access_token')
-    - |
-      curl -vX POST http://keycloak:8080/auth/admin/realms -d @keycloak/local-realm.json --header "Authorization: Bearer $TKN" --header "Content-Type: application/json"
+    - >
+      export TKN=$(curl -X POST
+        'http://keycloak:8080/auth/realms/master/protocol/openid-connect/token'
+        -H "Content-Type: application/x-www-form-urlencoded"
+        -d username=${KEYCLOAK_USER}
+        -d password=${KEYCLOAK_PASSWORD}
+        -d grant_type=password
+        -d client_id=admin-cli
+        | jq -r '.access_token')
+    - >
+      curl -vX POST http://keycloak:8080/auth/admin/realms
+        -d @keycloak/local-realm.json
+        -H "Authorization: Bearer $TKN"
+        -H "Content-Type: application/json"
 
-  build-project:
+  - name: build project
     image: quay.io/ukhomeofficedigital/openjdk11:v11.0.5_10
+    environment:
+      SPRING_PROFILES_ACTIVE: "development, local, test"
+      DB_HOST: postgres
+      AWS_LOCAL_HOST: localstack
     commands:
-      - export SPRING_PROFILES_ACTIVE='development, local, test'
-      - export DB_HOST='postgres'
-      - export KEYCLOAK_SERVER_ROOT=http://keycloak:8080
-      - export AWS_LOCAL_HOST='localstack'
       - ./gradlew build --no-daemon
-    when:
-      event: [push, pull_request, tag]
+    depends_on:
+      - wait for keycloak
 
-  sonar-scanner:
+  - sonar-scanner:
     image: quay.io/ukhomeofficedigital/sonar-scanner:v3.0.2
-    when:
-      event: [push, pull_request, tag]
+    depends_on:
+      - build project
 
-  docker-build:
-    image: docker:17.09.1
+  - name: build & push
+    image: plugins/docker
     environment:
-      - DOCKER_HOST=tcp://172.17.0.1:2375
-    commands:
-      - docker build -t hocs-info-service .
-    when:
-      branch: master
-      event: [push, tag]
+      DOCKER_PASSWORD:
+        from_secret: QUAY_ROBOT_PASSWORD
```

This has been changed in the others to a `TOKEN` as it's more accurate.

```suggestion
        from_secret: QUAY_ROBOT_TOKEN
```

---

It's probably less accurate, to be honest.